Libraries/LibCrypto/Hash/OpenSSLHashFunction.h

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2024, Altomani Gianluca <altomanigianluca@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/ByteString.h>
+#include <AK/NonnullOwnPtr.h>
+#include <AK/Types.h>
+#include <LibCrypto/Hash/HashFunction.h>
+
+#include <openssl/evp.h>
+
+namespace Crypto::Hash {
+
+template<typename T, size_t BlockS, size_t DigestS, typename DigestT = Digest<DigestS>>
+class OpenSSLHashFunction : public HashFunction<BlockS, DigestS, DigestT> {
+    AK_MAKE_NONCOPYABLE(OpenSSLHashFunction);
+
+public:
+    using HashFunction<BlockS, DigestS, DigestT>::update;
+
+    static NonnullOwnPtr<T> create()
+    {
+        auto* context = EVP_MD_CTX_new();
+        return make<T>(context);
+    }
+
+    explicit OpenSSLHashFunction(EVP_MD const* type, EVP_MD_CTX* context)
```

Comment:
two arg ctor doesn't need to be explicit. This should be only needed on the subclasses.